controllers/containerizedworkload_controller.go

```diff
@@ -0,0 +1,156 @@
+/*
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controllers
+
+import (
+	"context"
+	"github.com/pkg/errors"
+	"time"
+
+	cpv1alpha1 "github.com/crossplaneio/crossplane-runtime/apis/core/v1alpha1"
+	"github.com/go-logr/logr"
+
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	oamv1alpha2 "github.com/oam-dev/core-resource-controller/api/v1alpha2"
+)
+
+const (
+	oamReconcileWait = 30 * time.Second
+)
+
+// Reconcile error strings.
+const (
+	errRenderWorkload  = "cannot render workload"
+	errUpdateStatus    = "cannot apply status"
+	errApplyDeployment = "cannot apply the deployment"
+	errApplyService    = "cannot apply the service"
+	errGCDeployment    = "cannot clean up stale deployments"
+	errScaleDeployment = "cannot scale the deployment"
+)
+
+// ContainerizedWorkloadReconciler reconciles a ContainerizedWorkload object
+type ContainerizedWorkloadReconciler struct {
+	client.Client
+	Log    logr.Logger
+	Scheme *runtime.Scheme
+}
+
+// +kubebuilder:rbac:groups=core.oam.dev,resources=containerizedworkloads,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups=core.oam.dev,resources=containerizedworkloads/status,verbs=get;update;patch
+func (r *ContainerizedWorkloadReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
+	ctx := context.Background()
+	log := r.Log.WithValues("containerizedworkload", req.NamespacedName)
+	log.Info("Reconcile container workload")
```

Comment:
Do we really want this level of verbosity during normal reconciliation? Should these be debug logs?

Reply:
Sure

Reply:
see comments above, crossplane logger does not have Error logging.

Reply:
If not having error is a blocker you could still log these at "debug" (or at least below normal) level by using the `log.V(1).Info` convention of logr.